compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1917,6 +1925,40 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
     case _ =>
       refines.map(RefinedType(tp, _, _): Type).reduce(AndType(_, _))
 
+  // A relaxed version of isSubType, which compares method types
+  // under the standard arrow rule which is contravariant in the parameter types,
+  // but under the condition that signatures might have to match (see sigsOK)
+  // This relaxed version is needed to correctly compare dependent function types.
+  // See pos/i12211.scala.
+  def isSubInfo(info1: Type, info2: Type, symInfo1: Type, sigsOK: ((Type, Type) => Boolean) | Null = null, fallbackFn: ((Type, Type) => Boolean) | Null = null): Boolean = trace(i"isSubInfo $info1 <:< $info2"):
+    def fallback =
+      if fallbackFn != null then fallbackFn(info1, info2)
+      else isSubType(info1, info2)
+    val isCCPhase = ctx.phase == Phases.checkCapturesPhase
+
+    (info1, info2) match
```

Comment:
The previous version used two pattern matches on info1 and info2 instead of a pattern match on a tuple for performance, I suggest keeping the two pattern matches since this code path could be hot.